src/storage.ts

```diff
@@ -402,6 +320,236 @@ export class Storage {
 		}
 	}
 
+	/**
+	 * Download the source to the provided stream with a progress dialog.  Return
+	 * the status code or throw if the user aborts or there is an error.
+	 */
+	private async download(
+		client: AxiosInstance,
+		source: string,
+		writeStream: WriteStream,
+		headers?: AxiosRequestConfig["headers"],
+	): Promise<number> {
+		const baseUrl = client.defaults.baseURL;
+
+		const controller = new AbortController();
```

Comment:
I've known about this api for years and literally never seen someone use it in production. 🥹 so proud!!

Reply:
Ahaha I wish I could take credit but this was written by none other than Kyle himself (three years ago!  :open_mouth:)